lisa/sut_orchestrator/libvirt/platform.py

```diff
@@ -166,6 +165,19 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
                 parent_logger=get_logger("libvirt-platform"),
             )
 
+        vm_disk_img = self.runbook.requirement[type(self).type_name()]["disk_img"]
```

Comment:
@squirrelsc What are your thoughts on this?

Reply:
I found there is right usage in this file, please refer to below logic to get the `disk_img` in extended runbook.

```python
            node_runbook: BaseLibvirtNodeSchema = node_space.get_extended_runbook(
                self.__node_runbook_type(), type_name=type(self).type_name()
            )
```

Reply:
BTW, the value is in runbook already, it doesn't need to save to platform.

Reply:
@squirrelsc  Yes, `get_extended_runbook` is the right way. But `node_space` (node requirement) is not yet available for platform's `_initialize` to use. Hence, directly parsing the value from runbook.

Reply:
It sounds the vhd image should be a platform level config, instead of node level, because it needs to be loaded when initializing. Considering, if it needs to support different images in the same LISA run, how this value is loaded in initializing?

Reply:
VHD ideally should be node requirement. But, to reduce copying vhd multiple times to remote host, I made changes to copy only once in platform `_initialize`.
> Considering, if it needs to support different images in the same LISA run, how this value is loaded in initializing?

In general, isn't the testing image always constant in a single LISA run? Currently, there is no such use case to test different images in same LISA run.

Reply:
VHD cache can be controlled by a shared dict and locks. If a copied VHD exists, the vhd path can be used directly. If it's copying, other deployments can wait it by locks.
LISA run can use different images to run debug/release, normal VM/CVM together.

Reply:
Thanks, Chi, for the suggesting the approach. I'll make the changes. 
> LISA run can use different images to run debug/release, normal VM/CVM together.

Could you please point me to / provide an example of such way of running LISA?

Reply:
The vhd_img can be a variable, so it can work with combinator to test with more img. In the internal pipelines, there are examples, which uses the combinators.

Reply:
Oh ok. I was of the assumption that with combinators, every time a new platform object would be created. Thanks, I'll look into the examples.

Reply:
You are right, the combinator create a new lisa_runner and platform each time, so each platform is using the same img actually. But it doesn't need to rely on this assumption. We already find some limitations to use single image, in future, there may be hard requirements to use different images to deploy. In this PR, we can avoid setting img to platform.

Reply:
Made changes as suggested. @squirrelsc @cwize1 @anirudhrb  Please review. Thanks.